### PR TITLE
doc: fix create source commands in user docs

### LINKED ISSUE #2060: Fix create source command in simple file demo
see https://materialize.io/docs/sql/create-source/#creating-a-source-from-a-dynamic-unstructured-file

### PR DESCRIPTION
**Pull request by ruchirK**
Fixes #2060

### CODE REVIEW

**Comment by ruchirK** [comment]
I think I got the syntax right not totally sure about the style -- thats mainly what I'm asking about in this review

**Comment by ruchirK** [comment]
> tysm for catching these

it wasn't me!

File: doc/user/demos/log-parsing.md [hunk 1]
```diff
@@ -157,11 +157,10 @@ impose a structure on them:
 
 ```sql
 CREATE SOURCE requests
-FROM 'file:///log/requests'
-WITH (
-    regex='(?P<ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) - - \[(?P<ts>[^]]+)\] "(?P<path>(?:GET /search/\?kw=(?P<search_kw>[^ ]*) HTTP/\d\.\d)|(?:GET /detail/(?P<product_detail_id>[a-zA-Z0-9]+) HTTP/\d\.\d)|(?:[^"]+))" (?P<code>\d{3}) -',
-    tail=true
-    );
+FROM FILE '/log/requests'
+WITH ( tail=true )
```
Comment (benesch) [inline doc/user/demos/log-parsing.md]: Nikhil's Official Style Recommendation:

```sql
CREATE SOURCE requests
FROM FILE '/log/requests' WITH (tail = true)
FORMAG REGEX '...getwrecked...'
```

(at the very least I think you've got an erroneous trailing comma)

File: doc/user/sql/create-source.md [hunk 2]
```diff
@@ -386,10 +386,8 @@ offset, as well as the decoded value.
 ```sql
 CREATE SOURCE hex
 FROM FILE '/xxd.log'
-FORMAT REGEX '(?P<offset>[0-9a-f]{8}): (?:[0-9a-f]{4} ){8} (?P<decoded>.*)$'
-WITH (
-    tail=true
-);
+WITH ( tail = true )
```
Comment (benesch) [inline doc/user/sql/create-source.md]: `WITH (tail = true)`

File: doc/user/demos/log-parsing.md (hunk 1 above)
Comment (ruchirK) [reply doc/user/demos/log-parsing.md]: thanks for catching the comma

**Comment by ruchirK** [comment]
implemented Nikhils edits - am not going to go back and apply that style to the rest of the docs at this time

**Pull request merged**